scio-cosmosdb/src/main/scala/com/spotify/scio/cosmosdb/read/CosmosDbRead.scala

```diff
@@ -0,0 +1,49 @@
+package com.spotify.scio.cosmosdb.read
+
+import org.apache.beam.sdk.annotations.Experimental
+import org.apache.beam.sdk.annotations.Experimental.Kind
+import org.apache.beam.sdk.io.Read
+import org.apache.beam.sdk.transforms.PTransform
+import org.apache.beam.sdk.values.{PBegin, PCollection}
+import org.bson.Document
+import org.slf4j.LoggerFactory
+
+/** A {@link PTransform} to read data from CosmosDB Core (SQL) API. */
+@Experimental(Kind.SOURCE_SINK)
+private[cosmosdb] case class CosmosDbRead(
+  endpoint: String = null,
+  key: String = null,
+  database: String = null,
+  container: String = null,
+  query: String = null
+) extends PTransform[PBegin, PCollection[Document]] {
```

Comment:
It would be better to distinguish parameters in a dedicated case class model, and Transforms/Source/Reader in normal classes that passes the params only.
we should avoid the constructors with `this` argument

Reply:
Ok, excuse me, I was trying to mimic apache beam/java syntax, now I correct it.